File: 557.c
char* reverseWords(char* s) {
    char* begin,*end,*p;
    begin=end=p=s;
    char* res = (char*)malloc(strlen(s)+1);
    char* q= res;
    char* i;
    while(*p!='\0')
    {
        if(*p==' ')
        {
            end = p;          
            for(i=end-1;i>=begin;i--)
                *q++ = *i;
            *q++=' ';
            begin=end+1;
        }
        p++;
    }
    end = p;
    for(i=end-1;i>=begin;i--)
        *q++=*i;
    *q=*p;
    return res;
}

File: Python/557.py
class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        res = s.split()
        ans=""
        for temp in res:
            ans= ans +' '+temp[::-1]
        ans = ans.lstrip()
        return ans
           
    

res = Solution()
s="Let's take LeetCode contest"
s= res.reverseWords(s)
print(s)